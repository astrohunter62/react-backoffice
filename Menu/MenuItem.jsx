import React from 'react'
import PropTypes from 'prop-types'

import { ListItem, ListItemIcon, ListItemText } from 'material-ui/List'

const MenuItem = ({ redirectTo, url, title, icon: Icon }) => {
  return (
    <ListItem button onClick={() => redirectTo(url)}>
      {Icon? (
        <ListItemIcon>
          <Icon />
        </ListItemIcon>
      ) : null}
      <ListItemText primary={title} />
    </ListItem>
  )
}

MenuItem.propTypes = {
  redirectTo: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  icon: PropTypes.func,
}

export default MenuItem
