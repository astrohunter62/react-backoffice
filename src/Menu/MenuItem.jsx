import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import withStyles from 'material-ui/styles/withStyles'
import { ListItem, ListItemIcon, ListItemText } from 'material-ui/List'

const styles = {
  disabled: {
    opacity: 0.5,
    pointerEvents: 'none',
  },
}

const MenuItem = ({
  redirectTo,
  url,
  title,
  icon: Icon,
  disabled,
  classes,
}) => (
  <ListItem
    button
    onClick={() => redirectTo(url)}
    className={classNames({
        [classes.disabled]: disabled,
      })}
  >
    {Icon ? (
      <ListItemIcon>
        <Icon />
      </ListItemIcon>
      ) : null}
    <ListItemText primary={title} />
  </ListItem>
)

MenuItem.propTypes = {
  redirectTo: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  icon: PropTypes.func,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

MenuItem.defaultProps = {
  disabled: false,
  icon: null,
}

export default withStyles(styles)(MenuItem)
