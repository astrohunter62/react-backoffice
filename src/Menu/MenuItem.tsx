import React, { FunctionComponent } from "react";
import classNames from "classnames";
import {
  ListItem,
  ListItemIcon,
  ListItemText,
  makeStyles,
} from "@material-ui/core";

const useStyles = makeStyles(() => ({
  isDisabled: {
    opacity: 0.5,
    pointerEvents: "none",
  },
}));

type MenuItemProps = {
  redirectTo: (...args: any[]) => any;
  url: string;
  title: string;
  isDisabled?: boolean;
  icon?: JSX.Element;
  className?: string;
};

const MenuItem: FunctionComponent<MenuItemProps> = ({
  redirectTo,
  url,
  title,
  icon,
  isDisabled = false,
  className,
}) => {
  const classes = useStyles();

  return (
    <ListItem
      button
      onClick={() => redirectTo(url)}
      className={classNames(className, {
        [classes.isDisabled]: isDisabled,
      })}
    >
      {icon && <ListItemIcon>{icon}</ListItemIcon>}

      <ListItemText primary={title} />
    </ListItem>
  );
};

export default MenuItem;
