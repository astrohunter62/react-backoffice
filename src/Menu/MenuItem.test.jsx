import React from 'react'
import renderer from 'react-test-renderer'

import Icon from '@material-ui/icons/AccessAlarm'

import MenuItem from './MenuItem'

it('renders correctly', () => {
  const tree = renderer
    .create(<MenuItem
      title="Title"
      url="/"
      redirectTo={() => {}}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with icon', () => {
  const tree = renderer
    .create(<MenuItem
      title="Title"
      url="/"
      redirectTo={() => { }}
      icon={Icon}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly disabled', () => {
  const tree = renderer
    .create(<MenuItem
      title="Title"
      url="/"
      redirectTo={() => {}}
      isDisabled
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
