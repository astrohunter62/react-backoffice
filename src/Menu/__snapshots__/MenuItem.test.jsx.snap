// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MenuItem
  classes={
    Object {
      "isDisabled": "MenuItem-isDisabled-1",
    }
  }
  icon={null}
  isDisabled={false}
  redirectTo={[Function]}
  title="Title"
  url="/"
/>
`;

exports[`renders correctly disabled 1`] = `
<MenuItem
  classes={
    Object {
      "isDisabled": "MenuItem-isDisabled-1",
    }
  }
  icon={null}
  isDisabled={true}
  redirectTo={[Function]}
  title="Title"
  url="/"
/>
`;

exports[`renders correctly with icon 1`] = `
<MenuItem
  classes={
    Object {
      "isDisabled": "MenuItem-isDisabled-1",
    }
  }
  icon={
    Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "AccessAlarmIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    }
  }
  isDisabled={false}
  redirectTo={[Function]}
  title="Title"
  url="/"
/>
`;
